### PR TITLE
Add tests and fix bugs

### PR DESCRIPTION
**Pull request by yuval-si**

### CODE REVIEW

File: src/tests/test_amp.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import unittest
+from src.si import amp
+
+
+class TestAmp(unittest.TestCase):
+    def test_no_key(self):
+        with self.assertRaises(amp.AmpError) as ex:
+            amp.Amp("", "domains")
```
Comment (ajay-si) [inline src/tests/test_amp.py]: shouldn't this be amp_agents instead of domains?

File: src/tests/test_amp.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        self.assertEqual("'key' can't not be empty", str(ex.exception))
+
+    def test_no_domains(self):
```
Comment (ajay-si) [inline src/tests/test_amp.py]: test_no_key?

File: src/tests/test_amp.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
+import unittest
+from src.si import amp
+
+
+class TestAmp(unittest.TestCase):
+    def test_no_key(self):
```
Comment (ajay-si) [inline src/tests/test_amp.py]: test_no_amp_agents?

File: src/tests/test_amp.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+    def test_bad_amp_agent(self):
+        with self.assertRaises(amp.AmpError) as ex:
+            amp.Amp.parse_agent("bad_amp_agent")
+        self.assertEqual('bad amp agent bad_amp_agent', str(ex.exception))
+
+    def test_bad_protocol(self):
+        with self.assertRaises(amp.AmpError) as ex:
+            amp.Amp.parse_agent("ftp://ftp_server")
+        self.assertEqual("method in ftp://ftp_server must be 'http' ot 'https'", str(ex.exception))
```
Comment (walter-si) [inline src/tests/test_amp.py]: ot -> or

**Pull request merged**